### DISCUSSION TITLE
Controller for API and non-API route at the same time?

### DISCUSSION THREAD

**Discussion by fishnux** [adonisjs/core#1270] [Help] [answered]
Hi,

Suppose I'm building a traditional, server-side rendered website with a UserController. In the end of a method I would return a view.

Now I want to add a REST API. Should I use the same UserController? How would I prevent it from returning a view on the API route?

GET /user/1 <-- Returns a view
GET /api/user/1 <-- Should return JSON

What do you suggest?

**Comment by popovicieduard** [comment, marked as answer]
I think you can create a controller for both routes and return view or json response based on the header content-type or custom key, you could also add a middleware for your api routes that adds a header key.

https://preview.adonisjs.com/guides/http/content-negotiation
```
import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'

export default class PostsController {
  public async index ({ request, view }: HttpContextContract) {
    const posts = [{
      title: 'Adonis 101',
    }]

    switch (request.accepts(['html', 'json'])) {
      case 'html':
        return view.render('posts/index', { posts })
      case 'json':
        return posts
      default:
        return view.render('posts/index', { posts })
    }
  }
}
```

**Comment by RomainLanz** [comment]
Hey @fishnux! 👋 

It's up to you!

You can do whatever feels better in your mind. 😋 

Generally, you have two approaches:

1. You use the same controller and the same route and you use the [Content Negotation](https://preview.adonisjs.com/guides/http/content-negotiation#content-negotiation).
2. You create another controller in another namespace (like `Api/UserController.ts`).

**Comment by fishnux** [comment]
Thank you both!
